howso/engine/project.py

```diff
@@ -226,22 +235,22 @@ def delete(self) -> None:
         self._created = False
         self._id = None
 
-    def _update_attributes(self, project: BaseProject) -> None:
+    def _update_attributes(self, project) -> None:
         """
         Update the protected attributes of the project.
 
         Parameters
         ----------
-        project : BaseProject
+        project : Project
             The base project instance.
 
         Returns
         -------
         None
         """
-        for key in self.attribute_map.keys():
+        for key in Project.attribute_map.keys():
             # Update the protected attributes directly since the values
-            # have already been validated by the "BaseProject" instance
+            # have already been validated by the "Project" instance
             # and to prevent triggering an API update call
             setattr(self, f'_{key}', getattr(project, key))
```

Comment:
```suggestion
            setattr(self, f'_{key}', project.get(key))
```

Reply:
Should be addressed by: d66654ea10038f70a275bb4aa504128552745e60